third_party/blink/renderer/platform/audio/audio_destination.cc

```diff
@@ -100,6 +100,7 @@ AudioDestination::AudioDestination(AudioIOCallback& callback,
           AudioBus::Create(number_of_output_channels, render_quantum_frames)),
       callback_(callback),
       frames_elapsed_(0),
+      state_change_lock_("AudioDestination"),
```

Comment:
Curious:
* Does this make it ordered?
* And also: is there a `IsRecordingOrReplaying` guard inside the lock?